cpp/src/arrow/io/buffered_test.cc

```diff
@@ -329,7 +329,8 @@ class TestBufferedInputStream : public FileTestFixture<BufferedInputStream> {
     local_pool_ = MemoryPool::CreateDefault();
   }
 
-  void MakeExample1(int64_t buffer_size, MemoryPool* pool = default_memory_pool()) {
+  void MakeExample1(int64_t buffer_size, MemoryPool* pool = default_memory_pool(),
+                    bool fill_raw_read_bound = false) {
```

Comment:
We should instead pass the raw_read_bound explicitly, to allow setting it to a smaller value than the file size.